File: src/main/java/com/ebook/ebookapi/orden/modelo/Orden.java
package com.ebook.ebookapi.orden.modelo;

import com.ebook.ebookapi.book.modelo.Book;
import com.ebook.ebookapi.metodopago.modelo.MetodoPago;
import com.ebook.ebookapi.user.modelo.Usuario;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "orden")
public class Orden {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "monto_total", nullable = false, precision = 10, scale = 2)
    private BigDecimal montoTotal;

    @Column(nullable = false, length = 20)
    private String estado = "PENDIENTE";

    @Column(name = "fecha_creacion")
    private LocalDateTime fechaCreacion = LocalDateTime.now();

    @ManyToOne
    @JoinColumn(name = "id_usuario")
    private Usuario usuario;

    @ManyToOne
    @JoinColumn(name = "id_metodopago")
    private MetodoPago metodoPago;

    @ManyToOne
    @JoinColumn(name = "id_libro")
    private Book libro;
}

File: src/types/orden.ts
export interface Orden {
    id: number;
    montoTotal: number;
    estado: 'PENDIENTE' | 'APROBADO' | 'RECHAZADO';
    fechaCreacion: string;

    idUsuario: number;
    nombreUsuario : string;
    emailUsuario : string;

    idLibro: number;
    tituloLibro?: string;
    autorLibro: string;

    idMetodoPago: number;
    nombreMetodoPago?: string;
    
}

export interface OrdenRequest {
    idUsuario: number;
    idLibro: number;
    idMetodoPago: number;
    montoTotal: number;
}

export interface UpdateEstadoRequest {
    estado: string;
}
